Makefile

```
@@ -1,2 +1,12 @@
-badge.svg:
-	curl -o 'badge.svg' 'https://img.shields.io/badge/tested_with-Aqua.jl-05C3DD.svg?logo=data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiIHN0YW5kYWxvbmU9Im5vIj8+CjwhLS0gQ3JlYXRlZCB3aXRoIElua3NjYXBlIChodHRwOi8vd3d3Lmlua3NjYXBlLm9yZy8pIC0tPgoKPHN2ZwogICB3aWR0aD0iMTkiCiAgIGhlaWdodD0iMTQiCiAgIHZpZXdCb3g9IjAgMCA1LjAyNzA4MzIgMy43MDQxNjY4IgogICB2ZXJzaW9uPSIxLjEiCiAgIGlkPSJzdmc1IgogICBzb2RpcG9kaTpkb2NuYW1lPSJiYWRnZS5zdmciCiAgIGlua3NjYXBlOnZlcnNpb249IjEuMi4yIChiMGE4NDg2NTQxLCAyMDIyLTEyLTAxKSIKICAgeG1sbnM6aW5rc2NhcGU9Imh0dHA6Ly93d3cuaW5rc2NhcGUub3JnL25hbWVzcGFjZXMvaW5rc2NhcGUiCiAgIHhtbG5zOnNvZGlwb2RpPSJodHRwOi8vc29kaXBvZGkuc291cmNlZm9yZ2UubmV0L0RURC9zb2RpcG9kaS0wLmR0ZCIKICAgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIgogICB4bWxuczpzdmc9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KICA8c29kaXBvZGk6bmFtZWR2aWV3CiAgICAgaWQ9Im5hbWVkdmlldzEzIgogICAgIHBhZ2Vjb2xvcj0iI2ZmZmZmZiIKICAgICBib3JkZXJjb2xvcj0iIzAwMDAwMCIKICAgICBib3JkZXJvcGFjaXR5PSIwLjI1IgogICAgIGlua3NjYXBlOnNob3dwYWdlc2hhZG93PSIyIgogICAgIGlua3NjYXBlOnBhZ2VvcGFjaXR5PSIwLjAiCiAgICAgaW5rc2NhcGU6cGFnZWNoZWNrZXJib2FyZD0iMCIKICAgICBpbmtzY2FwZTpkZXNrY29sb3I9IiNkMWQxZDEiCiAgICAgc2hvd2dyaWQ9ImZhbHNlIgogICAgIGlua3NjYXBlOnpvb209IjI1LjI2MjExNiIKICAgICBpbmtzY2FwZTpjeD0iOC45MDY2MTcyIgogICAgIGlua3NjYXBlOmN5PSIyLjA3ODIxMDciCiAgICAgaW5rc2NhcGU6d2luZG93LXdpZHRoPSIxNDQ0IgogICAgIGlua3NjYXBlOndpbmRvdy1oZWlnaHQ9IjEyMzYiCiAgICAgaW5rc2NhcGU6d2luZG93LXg9IjQ1MyIKICAgICBpbmtzY2FwZTp3aW5kb3cteT0iMTcyIgogICAgIGlua3NjYXBlOndpbmRvdy1tYXhpbWl6ZWQ9IjAiCiAgICAgaW5rc2NhcGU6Y3VycmVudC1sYXllcj0ic3ZnNSIgLz4KICA8ZGVmcwogICAgIGlkPSJkZWZzMiIgLz4KICA8ZWxsaXBzZQogICAgIHN0eWxlPSJmaWxsOiMzODk4MjY7ZmlsbC1vcGFjaXR5OjE7c3Ryb2tlOiNmZmZmZmY7c3Ryb2tlLXdpZHRoOjAuMztzdHJva2UtbWl0ZXJsaW1pdDo0O3N0cm9rZS1kYXNoYXJyYXk6bm9uZTtzdHJva2Utb3BhY2l0eToxIgogICAgIGlkPSJwYXRoODU0IgogICAgIGN4PSIyLjUxODg3MTEiCiAgICAgY3k9IjEuNTIzODQxOSIKICAgICByeD0iMS4zMjI5MTYyIgogICAgIHJ5PSIxLjMyMjkxNjciIC8+CiAgPGNpcmNsZQogICAgIHN0eWxlPSJmaWxsOiNjYjNjMzM7ZmlsbC1vcGFjaXR5OjE7c3Ryb2tlOiNmZmZmZmY7c3Ryb2tlLXdpZHRoOjAuMztzdHJva2UtbWl0ZXJsaW1pdDo0O3N0cm9rZS1kYXNoYXJyYXk6bm9uZTtzdHJva2Utb3BhY2l0eToxIgogICAgIGlkPSJwYXRoMTAxMDMiCiAgICAgY3g9IjEuMTUzMjQ1OCIKICAgICBjeT0iMi41ODIxNzUiCiAgICAgcj0iMC45MjA4NTc3MyIgLz4KICA8Y2lyY2xlCiAgICAgc3R5bGU9ImZpbGw6Izk1NThiMjtmaWxsLW9wYWNpdHk6MTtzdHJva2U6I2ZlZmVmZTtzdHJva2Utd2lkdGg6MC4zO3N0cm9rZS1taXRlcmxpbWl0OjQ7c3Ryb2tlLWRhc2hhcnJheTpub25lO3N0cm9rZS1vcGFjaXR5OjEiCiAgICAgaWQ9InBhdGgxMDEwMy0zIgogICAgIGN4PSIzLjg4NDQ5NjUiCiAgICAgY3k9IjIuNTgyMTc1IgogICAgIHI9IjAuOTIwODU3NzMiIC8+CiAgPGcKICAgICBpZD0ibGF5ZXIyIgogICAgIHN0eWxlPSJkaXNwbGF5Om5vbmUiCiAgICAgdHJhbnNmb3JtPSJ0cmFuc2xhdGUoMC43OTM3NTA2MykiIC8+Cjwvc3ZnPgo='
+create_badge:
+	SVG_BASE64=$(shell base64 -w 0 docs/src/assets/logo.svg); \
+	curl -o "badge.svg" "https://img.shields.io/badge/tested_with-Aqua.jl-05C3DD.svg?logo=data:image/svg+xml;base64,$$SVG_BASE64"
+
+create_favicon:
```

Actually this target could also be turned into a real one, no?
```suggestion
favicon.ico: docs/src/assets/logo.svg
```

---

Sorry, I'm not much familiar with Makefile. Could you explain the suggestion in more detail?

---

The main purpose of `make` is to define rules how to create certain files (the "targets") from other files (the "dependencies" of the target) using certain rules. Then `make` uses file modification times to decide whether to regenerate the "target": if the target is missing, or if any of the dependencies is newer than the target, then it will run the rule to (re-)generate the target.

This is why my change proposes: it defines a target `facicon.ico` with a single dependency.

In contrast, the targets you added so far in `Makefile` are what is called "phony" target: e.g. the target `create_favicon` has no dependencies, and is simple meant to always be run when requested. When doing there, there really isn't much need for a `Makefile`, as then one could simply write a script which does the same job. (Phony targets can still be useful to support other regular targets without having to add additional script files, but here we are *only* defining phony targets.

In the end it is not super important, it just seems odd to not do this case with a "regular" target.

(BTW for `badge.svg` I did not propose a regular target; the reason is that it of course depends on whatever shields.io does, i.e., they might one day change how things look, and we can't detect that by inspecting a file locally).

---

Thank you for the detailed explanation! I will update the PR tomorrow.

---

I have updated `Makefile` as suggested, and also revert `create_badge` to `badge.svg`. I agree that the target depends on what shields.io does, but I think we don't need to check the update from shields.io every time. (We can check it via `make -B` if we need to update all generated files if we want.)

---

agreed